DancingGoat/Controllers/ControllerBase.cs

```diff
@@ -51,6 +58,13 @@ public static DeliveryClient CreateDeliveryClient()
             var options = provider.GetDeliveryOptions();
 
             options.ProjectId = ProjectUtils.GetProjectId();
+
+            if (ProjectUtils.IsInPreviewMode(httpContext) && !string.IsNullOrEmpty(ProjectUtils.GetPreviewApiKey(httpContext)))
+            {
+                options.UsePreviewApi = true;
+                options.PreviewApiKey = ProjectUtils.GetPreviewApiKey(httpContext);
```

Comment:
You are getting PreviewApiKey twice

Reply:
It's just consistency check, it's synchronized imo OK.